sympy/core/numbers.py

```diff
@@ -1816,13 +1816,18 @@ def _eval_power(self, expt):
                 # (4/3)**2 -> 4**2 / 3**2
                 return Rational(self.p**expt.p, self.q**expt.p, 1)
             if isinstance(expt, Rational):
-                if self.p != 1:
-                    # (4/3)**(5/6) -> 4**(5/6)*3**(-5/6)
-                    return Integer(self.p)**expt*Integer(self.q)**(-expt)
-                # as the above caught negative self.p, now self is positive
-                return Integer(self.q)**Rational(
-                expt.p*(expt.q - 1), expt.q) / \
-                    Integer(self.q)**Integer(expt.p)
+                intpart = expt.p // expt.q
+                if intpart:
+                    remfracpart = expt.q - expt.p
+                    if self.p != 1:
+                        return Integer(self.p)**expt*Integer(self.q)**Rational(remfracpart, expt.q)*Rational(1, self.q)
```

Comment:
Isn't this just `return Integer(self.p)**expt*Integer(self.q)**(-expt)`? or `Integer(self.p)**expt*Integer(self.q)**Rational(remfracpart - 1, expt.q)`?

Reply:
Note that the logic of `if intpart` was wrong in the code you commented on, so there is clearly a limited amount of test with a rational exponent > 1... (only the gamma-tests it seem).